solc/CommandLineInterface.cpp

```diff
@@ -856,13 +856,11 @@ bool CommandLineInterface::processInput()
 				return false;
 			}
 		}
-		if (optimize && inputLanguage == Input::Assembly)
+		if (optimize && inputLanguage != Input::StrictAssembly)
 		{
 			serr() <<
-				"Optimizer cannot be used for loose assembly. Use --" <<
+				"Optimizer cannot be used for both loose assembly and Yul. Use --" <<
```

Comment:
This might be interpreted as if the combination of loose assembly and yul cannot be used. What about "Can only be used for strict assembly?"

Reply:
fixed, yes the proposed change is more elegant